src/connect/connect.cpp

```diff
@@ -275,7 +275,16 @@ optional<OnlineStatus> connect::communicate(CachedFactory &conn_factory) {
 
     // Handle sleeping first. That one doesn't need the connection.
     if (auto *s = get_if<Sleep>(&action)) {
-        osDelay(s->milliseconds);
+        for (size_t i = 0; i < s->milliseconds / IDLE_WAIT; i++) {
+            // In case there's a change in situation during a long sleep, we
+            // want to retry sooner (new config might lead to being able to
+            // connect or something).
+            if (std::get<1>(printer.config(false))) {
+                return nullopt;
+            }
+        }
+
+        osDelay(s->milliseconds % IDLE_WAIT);
```

Comment:
Isn't this suppose to be in the loop and the amount slept be `IDLE_WAIT`, so we wait `s->miliseconds / IDLE_WAIT` times for `IDLE_WAIT` giving us total amount of `s->milliseconds` of sleeping? Seems to me, that we just check a bunch of times in a row and then wait for smaller amount of time. :thinking:

Reply:
Good catch!